docs/config.adoc

```diff
@@ -823,6 +823,45 @@ kanata -c startup.cfg -c 2nd.cfg -c 3rd.cfg
 Given the above startup command,
 activating `(lrld-num 2)` would reload the `2nd.cfg` file.
 
+==== Automatic file watching
+
+You can enable automatic reloading of configuration files when they change on disk
+by using the `--watch` command line flag. This requires building kanata with the
+`watch` feature enabled.
+
+[source]
+----
+kanata --cfg config.kbd --watch
+----
+
+When `--watch` is enabled:
+
+* Kanata monitors all configuration files specified with `--cfg` directly for changes
+* When a file is modified, kanata automatically triggers a live reload
+* Changes are debounced with a 500ms delay to avoid excessive reloading
+* Only the configuration files themselves are watched (not included files)
+
+**Limitations:**
+
+* The `--watch` flag is only available when kanata is compiled with the `watch` feature
```

Comment:
Remove this clause: `watch` should be a default feature with the option to exclude for technical users who want to minimize binary size for features they don't use.